### PR TITLE
Workflow

### PR DESCRIPTION
**Pull request by mattheww95**
Created an initial workflow and added basic tests, with some data. Linting will likely fail as I have not updated the `nextflow_schema.json` yet.

### CODE REVIEW

**Comment by mattheww95** [comment]
Linting tests are failing as I did not update the schema. I wanted prioritize getting the main "bones" of the pipeline in with test data so that critical feedback can be provided on the structure of the pipeline itself to make sure it fits what is expected.

File: workflows/gas_nomenclature.nf [hunk 1]
```diff
@@ -62,11 +78,47 @@ workflow GAS_NOMENCLATURE {
...
+
+    merged_queries = LOCIDEX_MERGE_QUERY(query_values, query_tag)
+    ch_versions = ch_versions.mix(merged_queries.versions)
+
+
+    // PROFILE DISTS processes
+
+    mapping_file = prepareFilePath(params.pd_mapping_file)
+    if(mapping_file == null){
+        exit 1, "${params.pd_mapping_file}: Does not exist but was passed to the pipeline. Exiting now."
```
Comment (emarinier) [inline workflows/gas_nomenclature.nf:97]: Do you know if these are the accepted way of exiting from nextflow, or should we use `error()`?

**Review by emarinier — CHANGES_REQUESTED** [review]
I think the changes here look good overall. I have some general comments:

- It might be worth adding failure / error testing.
- Generally, I'd advise against merging any TODOs into a dev / main branch, and instead do the TODO or make an issue on GitHub to do it later.
- I assume we're going forward with default parameters all listed in the `nextflow.config`?

**Comment by emarinier** [comment]
Oh, also getting the tests to pass.

**Comment by mattheww95** [comment]
Good news all tests passed! 

I am not proud of it, but to resolve the TODO I changed it to a point for discussion.

As for the defualt parameters, I dont know how most of the parameters to be passed to the tools are meant to be used so I just left them there for now.

I can add failure testing as well! I just thought it would be better to leave that as a seperate PR, I just wanted to get the initial skeleton of the tests implemented and full disclosure I was going to bug you about those afterwards...

**Review by emarinier — APPROVED** [review]

**Pull request merged**